### ISSUE TITLE
Publically accessible rspecs

### ISSUE THREAD

**Issue by innnkkki** [GENI-NSF/geni-portal#546]
It would be great if we could use the rspec database of the portal in other tools
(Flack, omni) so that we have a place where people can upload their rspecs. 

This means that we need some of the portal pages to be outside of hte autorization, so that anyone can access them. I can imagine that along with the public/private options we have now, we can have one more option "make this rspec publicly available". This can be an option for private or public rspecs. 

If we decide that this is too hard, maybe we can then allow users to use rspecs providing a URL. 

 _Imported from trac ticket [#546](http://trac.gpolab.bbn.com/proto-ch/ticket/546),  created by nriga on 05-18-2013 at 14:43, last modified: 04-08-2014 at 09:55_

**Comment by ahelsing** [comment]
The RSpecs are in a database. So to make them available requires a PHP page (or similar).  So there's work involved.

What would it mean for an RSpec that I mark private for me to check 'make this rspec publicly available'? It's publicly visible, just not automatically in everyone else's dropdown of RSpecs?

It is almost certainly easier for us to accept a URL for an RSpec. This will require a little thought when we start trying to do fancier stuff with RSpec management and guiding users in what aggregates work with an RSpec.

_Trac comment by ahelsing on 05-20-2013 at 08:58_

**Comment by innnkkki** [comment]
Replying to [ahelsing](comment:1):
> The RSpecs are in a database. So to make them available requires a PHP page (or similar).  So there's work involved.
I imagined that this won't come for free, I just wanted to make a ticket so we can track it and any decisions that we make
for this. 

> 
> What would it mean for an RSpec that I mark private for me to check 'make this rspec publicly available'? It's publicly visible, just not automatically in everyone else's dropdown of RSpecs?
Exactly. My thought was that I might was to have access to my rspecs from other tools, but I don't want this rspec to appear on the public list. 

> 
> It is almost certainly easier for us to accept a URL for an RSpec. This will require a little thought when we start trying to do fancier stuff with RSpec management and guiding users in what aggregates work with an RSpec.

It this is easy, then we can try adding a field in the add resources page. If on the back end the portal is using omni then indeed
this should be easy since omni already takes a URL. 


_Trac comment by nriga (github user: innnkkki) on 05-20-2013 at 10:01_

**Comment by seledwards** [comment]
> It is almost certainly easier for us to accept a URL for an RSpec. 

Could you rephrase this.  What does this mean?

> This will require a little thought when we start trying to do fancier stuff with RSpec management and guiding users in what aggregates work with an RSpec.

Also, don't understand this one.

Thanks,
Sarah


_Trac comment by sedwards (github user: seledwards) on 05-20-2013 at 10:27_

**Comment by ahelsing** [comment]
It is probably pretty easy for us to let the portal user specify the URL of an RSpec hosted elsewhere. And that is likely easier than making the RSpecs that we have in our database visible on some other web page without requiring authentication.

But we have talked about tagging RSpecs with attributes so that the portal can guide you - EG, an RSpec that specifies an ExoGENI disk image is not going to work at an InstaGENI AM. How will this functionality interact with the ability to specify an RSpec via a URL? Does the portal download it and auto tag it? Or just not provide any of these hints?

_Trac comment by ahelsing on 05-20-2013 at 10:36_

**Comment by innnkkki** [comment]
Replying to [ahelsing](comment:4):

> But we have talked about tagging RSpecs with attributes so that the portal can guide you - EG, an RSpec that specifies an ExoGENI disk image is not going to work at an InstaGENI AM. How will this functionality interact with the ability to specify an RSpec via a URL? Does the portal download it and auto tag it? Or just not provide any of these hints?

I don't know how these hints are going to be implemented, appear to the user, but if you could download it and provide these hints on the spot, that would be great. If this is too hard to implement, we can think about alternatives in this case. 


_Trac comment by nriga (github user: innnkkki) on 05-20-2013 at 10:39_

**Comment by seledwards** [comment]
Replying to [ahelsing](comment:4):
> It is probably pretty easy for us to let the portal user specify the URL of an RSpec hosted elsewhere. And that is likely easier than making the RSpecs that we have in our database visible on some other web page without requiring authentication.
> 
> But we have talked about tagging RSpecs with attributes so that the portal can guide you - EG, an RSpec that specifies an ExoGENI disk image is not going to work at an InstaGENI AM. How will this functionality interact with the ability to specify an RSpec via a URL? Does the portal download it and auto tag it? Or just not provide any of these hints?

I see.  You are proposing an alternative solution to the problem implicitly described in this ticket.  You are saying we should just do #464 "Allow loading RSpecs from a url" and leave it at that.

I agree that's a fine thing to do for the short term.  But I think we need to discuss and decide what we want to do in the long term (if anything).  Allowing the portal rspecs to be able to be publicly available is a fairly low effort way of creating a rudimentary rspec database which I think folks would find useful. It would be easy for people to add rspecs to it (just use your portal account) and we could easily have some of the indexing that you describe (which we want to do anyways).  These are two things we don't get from simply putting up a webserver.  So I think that's worth doing above and beyond merely doing #464, but that's just my opinion.

I don't have a strong opinion about timing.


_Trac comment by sedwards (github user: seledwards) on 05-20-2013 at 11:09_
